packages/web/src/scss/helpers/_links.scss

```diff
@@ -9,23 +9,22 @@
     cursor: cursors.$disabled;
 }
 
-// Allows link underline everywhere, except for headings
-.link-underlined {
+// Allows link underline everywhere, except for headings default state
+.link-underlined,
+[class*='typography-heading'] a:hover,
+[class*='typography-heading'] a:focus {
     text-decoration: underline;
+    text-decoration-skip-ink: none;
+    text-underline-offset: 0.1875em;
 }
 
 .link-stretched {
     @include links.stretch();
 }
 
-// All links in headings behave the same, underlined only in hover and active states
+// All links in headings behave the same
 [class*='typography-heading'] a {
```

Comment:
I only think our future version of Stylelint config is not going to like this because of `no-descending-specificity`. But it works either way.